src/Controls/src/Core/Handlers/Items/iOS/ObservableItemsSource.cs

```diff
@@ -279,7 +279,7 @@ internal int ItemsCount()
 		internal object ElementAt(int index)
 		{
 			if (_itemsSource is IList list)
-				return list[index];
+				return (index >= 0 && index < list.Count) ? list[index] : -1;
```

Comment:
This is the `ElementAt` call, so should return the object or null, not -1 (index).

I also see further down it is also returning -1, which is also quite wrong.

I think this code was copied from the `IndexOf` method below. 

But also, if we are getting here, there is something else wrong - the collection view is rendering a non-existent item? How did it get to this point?

Reply:
@mattleibow You are correct. Previously, we returned -1 based on the existing else part value. Now, we have changed the return value from -1 to null. Returning null is a more appropriate approach when a requested item does not exist in the ElementAt method. Please let us know if you have any concerns or additional feedback regarding this change.